packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/SystemCSharpMethodKinds.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    /// <summary>
+    /// This class contains the method kinds for the system C# methods. See <see cref="CSharpMethodKinds"/> .
+    /// </summary>
+    internal static class SystemCSharpMethodKinds
+    {
+        internal static readonly string JsonModelSerializationWrite = "JsonModelSerializationWrite";
```

Comment:
We are probably going down the path if every method needs to declare what kind it is.  The original intent was to be able to identify key types on output, and discriminate which builder should be called on input.

Lets leave these out for now and use an internal default of "Undefined" inside the csharpmethod ctor.

We can open a follow issue to track discussion around this going forward

Reply:
Opened https://github.com/microsoft/typespec/issues/3388 as a follow up for discussion.